pkg/core/fetch.go

```diff
@@ -226,7 +226,8 @@ func Fetch(ctx context.Context, storage database.Storage, pm *plugin.Manager, op
 	defer db.Close()
 	stateClient := state.NewClient(db, logging.NewZHcLog(&log.Logger, "fetch"))
 	// migrate CloudQuery core tables to latest version
-	if err := stateClient.MigrateCore(ctx, storage.DialectExecutor()); err != nil {
+	// context.DeadlineExceeded is handled inside runProviderFetch that returns a response and summary
+	if err := stateClient.MigrateCore(ctx, storage.DialectExecutor()); err != nil && !errors.Is(err, context.DeadlineExceeded) {
```

Comment:
I don't like this, but a better fix might require a bigger change in the code

Reply:
this will change in #731 but welcome to add the fix later

better fix needs a touch to `classifyError` maybe?

Reply:
>this will change in #731 but welcome to add the fix later

Thanks, I figured this will have to change soon. Happy to apply the fix later.

>better fix needs a touch to classifyError maybe?

Can you explain this a bit?
Do you mean add `DeadlineExceeded` (code and error) here https://github.com/cloudquery/cloudquery/blob/a3ecfc9166170e1bb77011befd11a5fbe1c86007/pkg/errors/error.go#L54 and https://github.com/cloudquery/cloudquery/blob/a3ecfc9166170e1bb77011befd11a5fbe1c86007/pkg/errors/error.go#L65

so they don't get reported?

Reply:
yes, `if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {`

Reply:
or just `if isErrorDeadlineExceeded(err)` with the addition below.

Reply:
since we have the isErrorDeadlineExceeded in the classifier this line/block change can be reverted (and not cause a conflict with my PR 😂 )

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/741/commits/cfb8b202b522fd453ad55dca2294967c64a90020

Reply:
I don't mind waiting with this after you merge your PR.
It's not a major issue and probably only happens in CI tests